android/sdk/src/main/java/com/taobao/weex/bridge/WXBridgeManager.java

```diff
@@ -2948,7 +2951,67 @@ public int callUpdateAttrs(String instanceId, String ref, HashMap<String, String
 
     return IWXBridge.INSTANCE_RENDERING;
   }
+  private boolean shouldReportGPULimit() {
+    IWXConfigAdapter adapter = WXSDKManager.getInstance().getWxConfigAdapter();
+    boolean report_gpu_limited_layout = false;
+    float sample_rate_of_report = 0;
+    if (adapter != null) {
+      try {
+        sample_rate_of_report = Float.parseFloat(adapter
+                .getConfig("android_weex_test_gpu",
+                        "sample_rate_of_report",
+                        "0"));
+      }catch(Exception e){
+        WXLogUtils.e(WXLogUtils.getStackTrace(e));
+      }
+      WXLogUtils.i("sample_rate_of_report : " + sample_rate_of_report);
+      if(Math.random() < sample_rate_of_report){
+        report_gpu_limited_layout = true;
+      }
+    }
+    return report_gpu_limited_layout;
+  }
 
+  private void reportIfReachGPULimit(String instanceId,String ref,GraphicSize layoutSize){
+    float limit = WXRenderManager.getOpenGLRenderLimitValue();
+    if(limit > 0 && (layoutSize.getHeight() > limit || layoutSize.getWidth() > limit)){
+      Map<String, String> ext = new ArrayMap<>();
```

Comment:
Change this line to Map<String, String> ext = new JSON object<>(), which will save you lot of work.